R/pkg/R/DataFrame.R

```diff
@@ -1304,24 +1306,62 @@ setClassUnion("characterOrColumn", c("character", "Column"))
 #' path <- "path/to/file.json"
 #' df <- jsonFile(sqlContext, path)
 #' arrange(df, df$col1)
-#' arrange(df, "col1")
 #' arrange(df, asc(df$col1), desc(abs(df$col2)))
+#' arrange(df, "col1")
+#' arrange(df, "col2", FALSE)
+#' arrange(df, "col1", decreasing=TRUE)
+#' arrange(df, "col1", "col2", c(TRUE, FALSE))
 #' }
 setMethod("arrange",
-          signature(x = "DataFrame", col = "characterOrColumn"),
+          signature(x = "DataFrame", col="Column"),
           function(x, col, ...) {
-            if (class(col) == "character") {
-              sdf <- callJMethod(x@sdf, "sort", col, list(...))
-            } else if (class(col) == "Column") {
               jcols <- lapply(list(col, ...), function(c) {
                 c@jc
               })
-              sdf <- callJMethod(x@sdf, "sort", jcols)
-            }
+
+            sdf <- callJMethod(x@sdf, "sort", jcols)
             dataFrame(sdf)
           })
 
 #' @rdname arrange
+#' @export
+setMethod("arrange",
+          signature(x = "DataFrame", col="character"),
+          function(x, col, ..., decreasing=FALSE) {
+
+            # all sorting columns
+            by <- list(col, ...)
+
+            # extracting the last element and uses it as decreasing if it is boolean
```

Comment:
remove this block of code. Sorting direction must be specified by "decreasing = ". No need to check if last element in "..." is decreasing or not.